frontend/test/resolution/testInteractive.cpp

```diff
@@ -129,9 +130,11 @@ computeAndPrintStuff(Context* context,
 
   for (const AstNode* child : ast->children()) {
     computeAndPrintStuff(context, child, inFn, calledFns,
-                         scopeResolveOnly, maxIdWidth);
-    if (child->isModule() || child->isFunction()) {
-      std::cout << "\n";
+                         scopeResolveOnly, maxIdWidth, quiet);
+    if (!quiet) {
+      if (child->isModule() || child->isFunction()) {
+        std::cout << "\n";
+      }
```

Comment:
Are you doing this for 'performance' measurement or for convenience of use? If it's the latter, consider threading through a "destination stream" (`std::cout` by default), and simply passing in a no-op stream ([my Googling showed me this](https://stackoverflow.com/questions/11826554/standard-no-op-output-stream)) if the output should be silenced. This way, you can avoid all this conditional logic that's broken up across multiple `if quiet` blocks. I don't think C++ is smart enough to ignore `<<` on a a "no-op stream", though, so it wouldn't help -- as much -- if you care about the performance hits of the prints.

Reply:
It's a little bit of both. I'm disinclined to switch to the no-op stream right now but it's an interesting trick to know about.